docs/cli-usage.md

```diff
@@ -0,0 +1,144 @@
+# Using the SOCI CLI
+
+This document provides instructions on how to use the SOCI CLI. And a comprehensive list of all the cli commands in one place.
+## Global flags
+
+- ```--address``` , ```-a```: Address for containerd's GRPC server
+- ```--namespace```, ```-n```: Namespace to use with commands
+- ```--timeout```: Timeout for commands
+- ```--debug```: Enable debug output
+- ```--content-store```: Use a specific content store (soci or containerd)
+
+## CLI commands
+- [soci create](#soci-create)
+- [soci push](#soci-push)
+- [soci rebuild_db](#soci-rebuild_db)
+- [soci index](#soci-index)
+- [soci ztoc](#soci-ztoc)
+
+
+### soci create
+Creates SOCI index for an image
+
+Output of this command is SOCI layers and SOCI index stored in a local directory
+>SOCI layer is named as \<image-layer-digest>.soci.layer
+>
+>SOCI index is named as \<image-manifest-digest>.soci.index
+
+Usage: ```soci create [flags] <image_ref> ```
+
+Flags:
+
+ - ```--spanSizeFlag``` : Span size that soci index uses to segment layer data. Default is 4MiB 
+ - ```--minLayerSizeFlag``` : Minimum layer size to build zTOC for. Smaller layers won't have zTOC and not lazy pulled. Default is 10MiB
+
+**Example:** 
+```
+soci create public.ecr.aws/soci-workshop-examples/ffmpeg:latest
+```
+
+### soci push
+Push SOCI artifacts to a registry by image reference.
+If multiple soci indices exist for the given image, the most recent one will be pushed.
+
+Usage: ```soci push [flags] <image_ref> ```
+
+> After pushing the soci artifacts, they should be available in the registry. 
+Soci artifacts will be pushed only
+> if they are available in the snapshotter's local content store.
+
+Flags:
+
+- ```--max-concurrent-uploads```: Max concurerent uploads. Default is 10
```

Comment:
nit: concurrent*

Also for the concurrency limit, I don't actually see this flag being referenced anywhere. Are we should this is implemented?

Reply:
These flags are defined in for the soci push command.

Ref : https://github.com/awslabs/soci-snapshotter/blob/7af8577ac627f991d561f94d476ea4529a02626a/cmd/soci/commands/push.go#L57